JavaShared/src/main/java/org/openzen/zenscript/javashared/prepare/JavaPrepareClassMethodVisitor.java

```diff
@@ -321,6 +324,10 @@ private void visitFunctional(FunctionalMember member, FunctionHeader header, Str
 
 		class_.empty = false;
 
-		class_.addMethod(member, method);
+		if (overrides.isPresent()) {
+			class_.addMethod(member, method, context.getJavaMethod(overrides.get()));
```

Comment:
Fixes `invokeVirtual` being called on the implementing class rather than on the interface class